service/src/main/java/org/cbioportal/service/impl/ClinicalAttributeServiceImpl.java

```diff
@@ -52,15 +59,19 @@ public ClinicalAttribute getClinicalAttribute(String studyId, String clinicalAtt
     @PreAuthorize("hasPermission(#studyId, 'CancerStudy', 'read')")
     public List<ClinicalAttribute> getAllClinicalAttributesInStudy(String studyId, String projection, Integer pageSize,
                                                                    Integer pageNumber, String sortBy,
-                                                                   String direction) {
+                                                                   String direction) throws StudyNotFoundException {
+
+        studyService.getStudy(studyId);
```

Comment:
Yeah it sucks that there is no way to indicate my purpose in this case other than comments (comments are my last resort, I try to make the code more clear by changing the code itself before adding comments), but yeah in this case it seems we need it, and Onur's question proves it. I'll add them.